### ISSUE TITLE
Command grouping with `{ }`

### ISSUE THREAD

**Issue by krocard** [fish-shell/fish-shell#10895]
TL;DR: add command grouping with `{}` (NOT `()`) to improve Fish efficiency and facilitate adoption

## Feature request
Command grouping using `{}` is a widely used feature of POSIX shells
- In error handling:
  - `command || { echo "failed; exit 1 }`
- To compose append command in a pipe chain:
  - `{ head -n1 csv; grep "key" csv } | nc localhost 123`
- To handle fallback:
  - `cmd1 && { cmd2 || cmd3 } && cmd4`

## Workaround
Fish offer the equivalent `begin; end`, nevertheless:
- it is long, which it is not convenient for the command line
- it is different than `{}` which shell users have been using for >30 years

## Backward compatibility
Fish is currently not using `{}` for anything other than [braces expansions](https://fishshell.com/docs/current/language.html#expand-brace) (just like POSIX) so there should be no backward compatibility issues.

## This issue does NOT advocate for `()`
Posix shell also uses `()` for command grouping, with the major difference that it spawns a subshell.
The author believes `()` not to be a good fit for Fish as:
- users rarely care about subshell, and most POSIX usage of `()` should be replaced by `{}`
  - subshell should not be encouraged, especially in Fish
- it would conflict with Fish command expansion (`()` <=> `$()`)
- it was already rejected in #8083

`{}` has none of those disadvantages.

**Comment by faho** [comment]
> it was already rejected in https://github.com/fish-shell/fish-shell/issues/8083

Which makes this a duplicate.

**Comment by krocard** [comment]
@faho `()` was rejected, not `{}`

**Comment by faho** [comment]
Ah, sorry, terrible font
